test/lit.cfg

```diff
@@ -153,24 +149,23 @@ class SwiftTest(lit.formats.ShTest, object):
                 test.config.environment["LLVM_PROFILE_FILE"] = os.devnull
                 self.skipped_tests.add(test.getSourcePath())
                 return
-            profdir = self.profdir_for_test(test)
-            if not os.path.exists(profdir):
-                os.makedirs(profdir)
 
-            test.config.environment["LLVM_PROFILE_FILE"] = \
-                os.path.join(profdir, "swift-%p.profraw")
+            _, tmp_base = lit.TestRunner.getTempPaths(test)
+            if self.coverage_mode == "NOT_MERGED":
+                profdir = tmp_base + '.profdir'
+                if not os.path.exists(profdir):
+                    os.makedirs(profdir)
+
+                test.config.environment["LLVM_PROFILE_FILE"] = \
+                    os.path.join(profdir, "swift-%p.profraw")
```

Comment:
Would a conflict arise if we named the test, say, `<test-source-file>-%p.profraw`? That would make it easier to identify which prof data corresponded to which test.